packages/lodestar/src/api/impl/validator/validator.ts

```diff
@@ -137,16 +136,12 @@ export class ValidatorApi implements IValidatorApi {
   public async getProposerDuties(epoch: Epoch): Promise<ProposerDuty[]> {
     await this.checkSyncStatus();
     assert.gte(epoch, 0, "Epoch must be positive");
-    const {state, epochCtx} = await this.chain.getHeadStateContext();
-    assert.lte(
-      epoch,
-      computeEpochAtSlot(this.config, state.slot) + 2,
-      "Cannot get duties for epoch more than two ahead"
+    assert.lte(epoch, this.chain.clock.currentEpoch + 2, "Cannot get duties for epoch more than two ahead");
+    const {state, epochCtx} = await this.chain.regen.getBlockSlotState(
+      this.chain.forkChoice.getHeadRoot(),
+      this.chain.clock.currentSlot
```

Comment:
actually, we can optimize stuff a bit here:
- if epoch > currentEpoch - process slots up to start slot of epoch
- else - use headState without slot processing

Reply:
I think we need to progress to at least the currentEpoch no matter what?
The optimization would be to not advance slots if the head block slot is already in currentEpoch.